src/main/java/org/wso2/securevault/secret/SecretManager.java

```diff
@@ -214,6 +191,51 @@ public void init(Properties properties) {
         initialized = true;
     }
 
+    /**
+     * Check whether to use the provider listed under secretRepositories
+     * or secretProviders property for resolving secrets
+     *
+     * @param secretAnnotation String contains the alias, the provider type and the repository type
+     * @return plain text value for the required secret
+     */
+    public String resolveSecret(String secretAnnotation) {
+
+        String[] secretAnnotationStrings = secretAnnotation.split(DELIMITER);
+        if (secretAnnotationStrings.length == 1) {
+            if (isLegacyProvidersExists) {
+                return getSecret(secretAnnotation);
+            }
+        }
+        return resolveSecret(secretAnnotationStrings);
+    }
+
+    /**
+     * Resolve the secret annotation for the secrets coming from
+     * repositories belongs to providers listed under secretProviders property
+     *
+     * @param annotation vaule retrieve by the resolveSecret as the value to be resolved
+     * @return If there is a secret , otherwise , alias itself
+     */
+    public String resolveSecret(String[] annotation) {
+
+        int length = annotation.length;
```

Comment:
why we need this variable

Reply:
Use to differentiate the secret annotation
if length == 1 -> provider type and the repository type should be assigned accordingly
if length == 3 -> annotation itself will provide the provider and the repository type